### ISSUE TITLE
[Bug]: braze.notifications.subscribeToUpdates process 4 events instead of 1 in IOS

### ISSUE THREAD

**Issue by SahharWithu** [braze-inc/braze-flutter-sdk#87]
### Which Platforms?

iOS

### Which Flutter Version?

3.22.2

### Which Braze Flutter SDK version?

11.0.0

### Repro Rate

100%

### Steps To Reproduce

braze.notifications.subscribeToUpdates process 4 events instead of one in IOS every time i send a push notification 

```

import UIKit
import Flutter
import BrazeKit
import braze_plugin
import BrazeLocation
import BrazeUI
import SDWebImage

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
    


  static var braze: Braze? = nil

  // These subscriptions need to be retained to be active
  var contentCardsSubscription: Braze.Cancellable?
  var pushEventsSubscription: Braze.Cancellable?
  var featureFlagsSubscription: Braze.Cancellable?

  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {

     
    let configuration = Braze.Configuration(
        apiKey: Configuration.brazeApiKey,
        endpoint: Configuration.brazeSdkEndpoint
    )
    configuration.sessionTimeout = 1
    configuration.triggerMinimumTimeInterval = 0
    configuration.location.automaticLocationCollection = true
    configuration.location.brazeLocationProvider = BrazeLocationProvider()

    if(Configuration.envirement == .dev) {
        configuration.logger.level = .debug
    } else {
        configuration.logger.level = .disabled
    }
    configuration.push.appGroup = Configuration.bundleIdentifier
    configuration.push.automation = true
    configuration.push.automation.requestAuthorizationAtLaunch = false

    let braze = BrazePlugin.initBraze(configuration)
      
    

    // - GIF support
    GIFViewProvider.shared = .sdWebImage

    // - InAppMessage UI
    let inAppMessageUI = CustomInAppMessagePresenter()
    braze.inAppMessagePresenter = inAppMessageUI

    // - Subscribe to various features and pass each model to the Dart layer
    contentCardsSubscription = braze.contentCards.subscribeToUpdates { contentCards in
      print("=> [Content Card Subscription] Received cards:", contentCards)
      BrazePlugin.processContentCards(contentCards)
    }
      
    pushEventsSubscription = braze.notifications.subscribeToUpdates { payload in
        print(
          """
          => [Push Event Subscription] Received push event:
             - type: \(payload.type)
             - title: \(payload.title ?? "<empty>"))
             - isSilent: \(payload.isSilent)
          """
        )
        BrazePlugin.processPushEvent(payload)
    }
      
      
    featureFlagsSubscription = braze.featureFlags.subscribeToUpdates { featureFlags in
      print("=> [Feature Flag Subscription] Received feature Flags:", featureFlags)
      BrazePlugin.processFeatureFlags(featureFlags)
    }
    
    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}



// MARK: In-app message UI

class CustomInAppMessagePresenter: BrazeInAppMessageUI {

  override func present(message: Braze.InAppMessage) {
    print("=> [In-app Message] Received message from Braze:", message)

    // Pass in-app message data to the Dart layer.
    BrazePlugin.processInAppMessage(message)

    // If you want the default UI to display the in-app message.
    super.present(message: message)
  }

}

// MARK: GIF support

extension GIFViewProvider {
  public static let sdWebImage = Self(
    view: { SDAnimatedImageView(image: image(for: $0)) },
    updateView: { ($0 as? SDAnimatedImageView)?.image = image(for: $1) }
  )
  private static func image(for url: URL?) -> UIImage? {
    guard let url = url else { return nil }
    return url.pathExtension == "gif"
      ? SDAnimatedImage(contentsOfFile: url.path)
      : UIImage(contentsOfFile: url.path)
  }
}
  
// MARK: Linking
  
extension AppDelegate {
  
  private func forwardURL(_ url: URL) {
    guard let controller: FlutterViewController = window?.rootViewController as? FlutterViewController else { return }
    let deepLinkChannel = FlutterMethodChannel(name: "deepLinkChannel", binaryMessenger: controller.binaryMessenger)
    deepLinkChannel.invokeMethod("receiveDeepLink", arguments: url.absoluteString)
  }
  
  // Custom scheme
  // See https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app for more information.
  override func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
    forwardURL(url)
    return true
  }
  
  // Universal link
  // See https://developer.apple.com/documentation/xcode/allowing-apps-and-websites-to-link-to-your-content for more information.
  override func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool {
    guard userActivity.activityType == NSUserActivityTypeBrowsingWeb,
          let url = userActivity.webpageURL else {
      return false
    }
    forwardURL(url)
    return true
  }
}
```

### Expected Behavior

only one event should be processed 

### Actual Incorrect Behavior

every time i send a notification to ios 4 events processed instead of 1

### Verbose Logs

```shell
=> [Push Event Subscription] Received push event: <BRZNotificationsPayload: 0x300057ed0>
=> [Push Event Subscription] Received push event: <BRZNotificationsPayload: 0x300057f20>
=> [Push Event Subscription] Received push event: <BRZNotificationsPayload: 0x300078d20>
=> [Push Event Subscription] Received push event: <BRZNotificationsPayload: 0x300044910>
```

**Comment by hokstuff** [comment]
Hi @SahharWithu,

According to the verbose logs you linked, the pointer address of the payload is a different address for all 4 instances. Can you confirm whether or not the fields are exactly the same? Based on your code snippet pasted above, I'd expect it to print out more than just the object address:
```
          => [Push Event Subscription] Received push event:
             - type: \(payload.type)
             - title: \(payload.title ?? "<empty>"))
             - isSilent: \(payload.isSilent)
```

I have a few additional questions:
1. Are you seeing a `payload.type` with both a value `.received` and `.opened`? On the underlying Swift SDK, there are separate push events when you initially receive the notification and when you actually click on the notification.
2. Are you testing on an iOS 18 device? In version [10.3.0 of the underlying Swift SDK](https://github.com/braze-inc/braze-swift-sdk/releases/tag/10.3.0), we added additional safeguards from processing the same notification multiple times. The most recent Flutter SDK version uses 10.2.0, so we will start the process to update this in case that remedies your issue.
3. Are you triggering this closure in `subscribeToUpdates` while the app is in the foreground or the background? Can you share the push payloads you are using?

To more easily send over the answers to these questions along with relevant code snippets and payloads, feel free to contact support@braze.com. Thank you!

**Comment by SahharWithu** [comment]
1 - all events are of type received 
```
=> [Push Event Subscription] Received push event:
   - type: received
   - title: test)
   - isSilent: false
=> [Push Event Subscription] Received push event:
   - type: received
   - title: test)
   - isSilent: false
=> [Push Event Subscription] Received push event:
   - type: received
   - title: test)
   - isSilent: false
=> [Push Event Subscription] Received push event:
   - type: received
   - title: test)
   - isSilent: false
   ```
2 - Yes im using ios 18 with xcode 16 and im using flutter sdk 
 
<img width="641" alt="Screenshot 2024-09-26 at 09 18 44" src="https://github.com/user-attachments/assets/961b6c42-7e7b-45b0-be7d-27f154b3bdb8">

3 - foreground , im using standard push in braze dashboard with content-mutable and content-available flags checked

**Comment by hokstuff** [comment]
1. Have you implemented both ["Automatic push handling" as well as "Manual push handling"](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/b1-standard-push-notifications)? Can you link any other iOS push-related code (e.g. `UIApplicationDelegate` or `UNUserNotificationCenterDelegate ` methods)?
2. It sounds like you are creating a visible push notification that also contains the `content-available` flag, [which Apple generally recommends to specifically be used by non-visible (silent) push](https://developer.apple.com/documentation/usernotifications/generating-a-remote-notification#:~:text=notification%20types.-,content%2Davailable,-Number) ([more info here](https://developer.apple.com/documentation/usernotifications/pushing-background-updates-to-your-app#Create-a-background-notification)). If you remove either the `content-available:1` flag from the payload or the visible text in the `title`, `subtitle`, or `body`, does that lower the amount of times the subscription is triggered?

In either case, given you are using iOS 18, we are planning to release a new version that includes Swift SDK 10.3.0+ which should help alleviate some or all of the duplicate subscription issues. We will be in touch once this is released so you can teset it out.

**Comment by SahharWithu** [comment]
Hi @hokstuff 
im only using automatic push handling , full app delegate is included in ticket description 
will wait for next release .

**Comment by hokstuff** [comment]
Hi @SahharWithu,

We've just released [Flutter SDK 11.1.0](https://github.com/braze-inc/braze-flutter-sdk/releases/tag/11.1.0) which includes the changes mentioned above around adding safeguards to prevent processing the same push payload more than once. Let us know if this resolves your issues!
